iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -573,10 +746,73 @@ ReceiverPortType::MemberType_t* PortManager::acquireReceiverPortData(const capro
     }
 }
 
+cxx::expected<PublisherPortRouDiType::MemberType_t*, PortPoolError>
+PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
+                                      const uint64_t& historyCapacity,
+                                      const ProcessName_t& processName,
+                                      mepoo::MemoryManager* payloadMemoryManager,
+                                      const RunnableName_t& runnable [[gnu::unused]], // @todo Fix introspection
+                                      const PortConfigInfo& portConfigInfo) noexcept
+{
+    /// @todo unique check only required for single producer type
+    // // check if already in list, we currently do not support multi publisher for one CaPro ID
+    // for (auto publisherPortData : m_portPool->getPublisherPortDataList())
+    // {
+    //     PublisherPortRouDiType publisherPort(publisherPortData);
+    //     if (service == publisherPort.getCaProServiceDescription())
+    //     {
+    //         LogWarn() << "Process '" << processName
+    //                   << "' tried to register an unique PublisherPort which is already used by '"
+    //                   << publisherPortData->m_processName << "' with service '"
+    //                   << service.operator cxx::Serialization().toString() << "'.";
+
+    //         if (publisherPort.isUnique())
+    //         {
+    //             errorHandler(Error::kPOSH__PORT_MANAGER_PUBLISHERPORT_NOT_UNIQUE, nullptr, ErrorLevel::MODERATE);
+    //             return cxx::error<PortPoolError>(PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS);
+    //         }
+    //         else
+    //         {
+    //             break;
+    //         }
+    //     }
+    // }
```

Comment:
Not a big fan of this huge out-commented section. Do you promise to fix it asap in a follow-up PR? ;-)

Reply:
Remember, when we discussed it we agreed to add `@todo #25` to everything that'll be cleaned-up by the follow-up PR, which will be done right after this one is merged. So it will stay just a couple of days.

Reply:
Fixed